test/unit_tests/logging/test_logging_cvode.cpp

```diff
@@ -0,0 +1,195 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): David J. Gardner @ LLNL
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Test logging output in CVODE
+ * ---------------------------------------------------------------------------*/
+
+#include <cmath>
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+
+// Include desired integrators, vectors, linear solvers, and nonlinear solvers
+#include "cvode/cvode.h"
```

Comment:
The CVODES version of this essentially only differs in which header is used. You could factor this shared code into a separate file `test_logging_cvode_shared.h`, removing the CVODE(S) include.
I think this file would simply be
```
#include "cvode/cvode.h"
#include "test_logging_cvode_shared.h"
```
while the CVODES one would be
```
#include "cvodes/cvodes.h"
#include "test_logging_cvode_shared.h"
```
That would save nearly 400 lines of duplicated code across the CVODE and IDA tests. x-macros can be used to have the print statements for the method name be different if desired.

Reply:
On second thought x-macros are probably not needed. Instead I suspect something like this works
```
#include "cvode/cvode.h"
static constexpr auto NAME = "CVODE";
#include "test_logging_cvode_shared.h"
```
and the shared header can use `NAME`.

Reply:
I'd like to keep them separate. The intent is expanded the CVODES version (and similarly for IDA) to cover the FSA and ASA cases in the future (along with improvements to the logging output with FSA and ASA).